app/src/main/java/com/example/playlistmaker/SearchActivity.kt

```diff
@@ -1,124 +1,119 @@
 import androidx.recyclerview.widget.RecyclerView
-import com.bumptech.glide.Glide
-import com.bumptech.glide.load.resource.bitmap.RoundedCorners
-import kotlin.random.Random
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
 
 
 class SearchActivity : AppCompatActivity() {
 
-
-
+    private val iTunesBaseUrl = "https://itunes.apple.com"
     private var searchText: String = ""
 
-    override fun onSaveInstanceState(outState: Bundle) {
-        super.onSaveInstanceState(outState)
-        outState.putString(SEARCH_TEXT_KEY, searchText)
-    }
+    private val retrofit = Retrofit.Builder()
+        .baseUrl(iTunesBaseUrl)
+        .addConverterFactory(GsonConverterFactory.create())
+        .build()
 
-    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
-        super.onRestoreInstanceState(savedInstanceState)
-        searchText = savedInstanceState.getString(SEARCH_TEXT_KEY, "")
-        findViewById<EditText>(R.id.search_input).setText(searchText)
-    }
+    private val iTunesService = retrofit.create(iTunesAPI::class.java)
+    private val tracks = ArrayList<Track>()
+    private val adapter = TrackAdapter(tracks)
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_search)
 
+        // Инициализация UI элементов
+        val searchInput = findViewById<EditText>(R.id.search_input) //Поле ввода
+        val clearButton = findViewById<ImageView>(R.id.clear_button)//Кнопка отчистки поля ввода
+        val backButton = findViewById<ImageView>(R.id.btn_settings_back)//Возврат на пред. страницу
+        val recyclerView = findViewById<RecyclerView>(R.id.recyclerView)//Список треков
+        val placeholderButton = findViewById<Button>(R.id.placeholderButton)//ТЕСТ ДЛЯ ПРОВЕРКИ
 
-        val searchInput = findViewById<EditText>(R.id.search_input)
-        val clearButton = findViewById<ImageView>(R.id.clear_button)
-        val backButton = findViewById<ImageView>(R.id.btn_settings_back)
-
-        //
-
-        val recyclerView = findViewById<RecyclerView>(R.id.recyclerView)
         recyclerView.layoutManager = LinearLayoutManager(this)
+        recyclerView.adapter = adapter
+
+
+        // Установка слушателя для выполнения поиска при нажатии на клавиатуре "Done"
+        searchInput.setOnEditorActionListener { _, actionId, _ ->
+            if (actionId == EditorInfo.IME_ACTION_DONE) {
+                // Выполнение поискового запроса
+                iTunesService.search(searchInput.text.toString()).enqueue(object :
+                    Callback<TrackResponse> {
+                        override fun onResponse(
+                            call: Call<TrackResponse>,
+                            response: Response<TrackResponse>
+                        ) {
+                            if (response.code() == 200) {
```

Comment:
```suggestion
                            if (response.isSuccessfull) {
```
Лучше воспользоваться такой проверкой, т.к. не только код 200 говорит об успешном выполнении запроса